### PR TITLE
New tests added. New logger. Refactoring.

### PR DESCRIPTION
**Pull request by Devas**
New tests added.
New logger for statistics.
Custom formatter for logger.
Refactoring.

### CODE REVIEW

File: crawler/src/main/java/com/javaacademy/crawler/jsoup/MatrasScrapper.java [hunk 1]
```diff
@@ -108,43 +99,20 @@ String getSmallThumbnail() {
...
-            return price;
-        }
+        return elements.isEmpty() ? 0 :
+                parsePrice(elements.text().replace("\u00a0zł", "").replace(",", "."));
     }
 
     @SuppressWarnings("Duplicates")
     @Override
     double getRetailPrice() {
         Elements elements = getDoc().select("div.this-main-price");
```
Comment (patkkka) [inline crawler/src/main/java/com/javaacademy/crawler/jsoup/MatrasScrapper.java]: You can go with select(".this-main-price") to be dependent only on class.

File: crawler/src/main/java/com/javaacademy/crawler/jsoup/MatrasScrapper.java [hunk 2]
```diff
@@ -108,43 +99,20 @@ String getSmallThumbnail() {
...
-            double price = 0;
-            try {
-                price = Double.parseDouble(elements.text().replace("\u00a0zł", "").replace(",", "."));
-            } catch (NumberFormatException e) {
-                AppLogger.logger.log(Level.WARNING, "Exception while parsing, ", e);
-            }
-            return price;
-        }
+        return elements.isEmpty() ? 0 :
+                parsePrice(elements.text().replace("\u00a0zł", "").replace(",", "."));
```
Comment (patkkka) [inline crawler/src/main/java/com/javaacademy/crawler/jsoup/MatrasScrapper.java]: .attr("content") gives you number without unit and with .

File: crawler/src/test/java/com/javaacademy/crawler/jsoup/MatrasScrapperTests.java [hunk 3]
```diff
@@ -13,16 +15,24 @@
...
+        Assert.assertEquals(matrasScrapper.parseSinglePage(LINK).getTitle(), TITLE);
+    }
+
+    public void testScrapeNoPages() {
+        MatrasScrapper matrasScrapper = new MatrasScrapper();
+        matrasScrapper.setPagesToScrap(0);
+        Assert.assertEquals(matrasScrapper.scrape(), Collections.emptySet());
+    }
+
+    public void testScrapeSomePages() {
```
Comment (patkkka) [inline crawler/src/test/java/com/javaacademy/crawler/jsoup/MatrasScrapperTests.java]: Test methods could have better names without "test"

File: crawler/src/main/java/com/javaacademy/crawler/jsoup/MatrasScrapper.java (hunk 2 above)
Comment (Devas) [reply crawler/src/main/java/com/javaacademy/crawler/jsoup/MatrasScrapper.java]: :+1:

**Pull request merged**